gitprovider/repositoryref.go

```diff
@@ -83,6 +83,11 @@ type Slugger interface {
 	Slug() string
 }
 
+// URLClone returns the HTTPS URL to clone the repository.
+type URLClone interface {
```

Comment:
Not sure why you opted for `URLClone` instead of `CloneURL` (used elsewhere?)

In any case, I think naming it something like the following would read better:

```suggestion
type CloneableURL interface {
   GetCloneURL(prefix string, transport TransportType)
}
```